From 9bb8c6969f5c3b4b12cf701e3c7c8f760caee0f0 Mon Sep 17 00:00:00 2001
From: CI Assistant <yossi.ilkanaev@intel.com>
Date: Fri, 26 Oct 2018 12:16:10 +0200
Subject: [PATCH] Merge pull request #232 in SW_UGW/linux from
 feature/UGW_SW-27826-code-changes-for-queue-length-to-8.x to master

* commit '2b30dc5a6262867eab160172e923f8bc002598b1':
  UGW_SW-27826: Address review comments
  UGW_SW-27826: Added macros and structure variables for Queue Length feature
---
 include/uapi/net/ppa_api.h | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/include/uapi/net/ppa_api.h b/include/uapi/net/ppa_api.h
index 7b91d13018b6..aea7f0df4653 100755
--- a/include/uapi/net/ppa_api.h
+++ b/include/uapi/net/ppa_api.h
@@ -412,6 +412,17 @@
 /*!
 	\brief PPA_F_PPPOATM */
 #define PPA_F_PPPOATM				0x80000000
+
+/*
+ * Queue Threshold Color Drop Flags
+ */
+/** PPA_QOS_DT_F_GREEN_THRESHOLD:Set Green color threshold drop tail flag */
+#define PPA_QOS_DT_F_GREEN_THRESHOLD 0x00000001
+/** PPA_QOS_DT_F_YELLOW_THRESHOLD:Set Yellow color threshold drop tail flag*/
+#define PPA_QOS_DT_F_YELLOW_THRESHOLD 0x00000002
+/** PPA_QOS_DT_F_RED_THRESHOLD:Set Red color threshold drop tail flag*/
+#define PPA_QOS_DT_F_RED_THRESHOLD 0x00000004
+
 /*
  * internal flag
  */
@@ -1545,6 +1556,7 @@ typedef struct {
 	uint32_t max_th1; /*!< Max Threshold for WRED Curve 1 in % of
 						qlen*/
 	uint32_t max_p1; /*!< Max Drop Probability % at max_th1 for WRED Curve 1 */
+	uint32_t drop_th1; /*!< Threshold for red color packet*/
 } PPA_QOS_WRED;
 /*!
 	\brief QoS Queue Drop configuration structure
@@ -1554,6 +1566,7 @@ typedef struct {
 	PPA_QOS_DROP_MODE mode; /*!< Mode of Queue Drop - Taildrop, WRED
 							 */
 	PPA_QOS_WRED wred; /*!< WRED configuration of the queue */
+	int32_t flags;  /*!< Drops flags - Drop for red color, yellow color, red color */
 } PPA_QOS_DROP_CFG;
 /*!
 	\brief Engine Selector Capabilities Structure
